hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -345,6 +363,28 @@ short getFlags() {
         return flags;
     }
 
+
+    /**
+     * Called when an <tt>Exception</tt>/<tt>Error</tt> is thrown during operation execution.
+     * <p/>
+     * By default this method does nothing.
+     * Operation implementations can override this behaviour due to their needs.
+     * <p/>
+     * This method is called on node & thread that's executing the operation.
+     *
+     * @param e Exception/Error thrown during operation execution
+     */
+    public void onFailure(Throwable e) {
```

Comment:
The problem is that we add more capabilities to the operation. Can't we do this by an interface? So having an interface FailureCallbackOperation with a single method 'onFailure'. This way you keep the operation nice and clean.